airflow/triggers/base.py

```diff
@@ -124,6 +124,11 @@ def __repr__(self) -> str:
         classpath, kwargs = self.serialize()
         return self.repr(classpath, kwargs)
 
+    def __eq__(self, other: Any) -> bool:
+        if not isinstance(other, BaseTrigger):
+            return False
```

Comment:
```suggestion
        if not isinstance(other, BaseTrigger):
            return NotImplemented
```
It gives the `other` a chance to have it's `__eq__` method called.